src/toil/job.py

```diff
@@ -289,7 +295,8 @@ def wrapJobFn(fn, *args, **kwargs):
         :rtype: toil.job.JobFunctionWrappingJob
         """
         if PromisedRequirement.convertPromises(kwargs):
-            return PromisedRequirementJobFunctionWrappingJob(fn, *args, **kwargs)
+            # See PromisedRequirementFunctionWrappingJob
```

Comment:
Actually, you can refactor this such that .encapsulate() is only used once. Then you only have to explain it only once. After the refactoring, this line below would look like this:

```
return PromisedRequirementJobFunctionWrappingJob.create(fn, *args, **kwargs)
```

That's called a factory method.